app/models/project.rb

```diff
@@ -38,6 +38,7 @@ class Project < ApplicationRecord
   validate :tutorial_enrolment_same_campus, if: :will_save_change_to_enrolled?
 
   after_update :check_withdraw_from_groups, if: :saved_change_to_enrolled?
+  after_update :update_task_stats, if: :saved_change_to_target_grade?
```

Comment:
**thought:** This might increase latency when users try to save the target grades. It could be premature optimisation but we can move this to an async job if it is ever becomes an issue.

Reply:
For an individual project, the update of task stats takes negligible time - so it should be ok. I'll add a to-do so that we can note this for future performance optimisations.